lib/qless/worker/base.rb

```diff
@@ -15,13 +15,16 @@ module Workers
 
     class BaseWorker
       attr_accessor :output, :reserver, :log_level, :interval, :paused,
-                    :options
+                    :options, :sighup_handler
 
       def initialize(reserver, options = {})
         # Our job reserver and options
         @reserver = reserver
         @options = options
 
+        # SIGHUP handler
+        @sighup_handler = options.fetch(:sighup_handler) { lambda { } }
```

Comment:
It would be nice to have a more useful default than a no-op but I can't think of what that would be.